src/sql/SqlService.ts

```diff
@@ -0,0 +1,290 @@
+            if (longValue.lessThanOrEqual(Long.fromInt(-2))) {
+                throw new RangeError('Timeout millis cannot be less than -1');
+            }
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('cursorBufferSize') && tryGetNumber(sqlStatementOptions.cursorBufferSize) <= 0) {
+            throw new RangeError('Cursor buffer size cannot be negative');
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('expectedResultType'))
+            tryGetEnum(SqlExpectedResultType, sqlStatementOptions.expectedResultType);
+
+        if (sqlStatementOptions.hasOwnProperty('returnRawResult'))
+            tryGetBoolean(sqlStatementOptions.returnRawResult);
+    }
+
+    /**
+     * Converts datetime related wrapper classes to string to be able to serialize them. (No default serializers yet)
+     * @internal
+     * @param value
+     */
+    convertToStringIfDatetimeValue(value: any) {
+        if (value instanceof HzLocalTime || value instanceof HzLocalDate || value instanceof HzLocalDateTime) {
+            return value.toString();
+        } else if (value instanceof HzOffsetDateTime) {
+            return value.toISOString();
+        } else {
+            return value;
+        }
+    }
+
+    execute(sql: SqlStatement): SqlResult;
+    execute(sql: string, params?: any[], options?: SqlStatementOptions): SqlResult;
+    execute(sql: string | SqlStatement, params?: any[], options?: SqlStatementOptions): SqlResult {
+        let sqlStatement: SqlStatement;
+
+        if (typeof sql === 'string') {
+            sqlStatement = {
+                sql: sql
+            };
+            if (params !== undefined) { // todo: document this
+                sqlStatement.params = params;
+            }
+            if (options !== undefined) {
+                sqlStatement.options = options;
+            }
+        } else if (typeof sql === 'object') { // assume SqlStatement is passed
+            sqlStatement = sql;
+        } else {
+            throw new IllegalArgumentError('Sql can be an object or string');
+        }
+
+        try {
+            SqlServiceImpl.validateSqlStatement(sqlStatement);
+        } catch (error) {
+            throw new IllegalArgumentError(`Invalid argument given to execute(): ${error.message}`, error)
+        }
+
+        const connection = this.connectionRegistry.getRandomConnection(true);
+        if (connection === null) {
+            throw new HazelcastSqlException(
+                this.connectionManager.getClientUuid(),
+                SqlErrorCode.CONNECTION_PROBLEM,
+                'Client is not currently connected to the cluster.'
+            );
+        }
+
+        const queryId = SqlQueryId.fromMemberId(connection.getRemoteUuid());
+
+        const expectedResultType: SqlExpectedResultType =
+            sqlStatement.options?.expectedResultType ? SqlExpectedResultType[sqlStatement.options.expectedResultType]
```

Comment:
I am not really familiar with typescript but what will happen if the `expectedResultType` has a 0 value? Which branch we will take?

Reply:
It takes strings of the enum, not values like 0 and 1.